src/order/compactly_generated.lean

```diff
@@ -470,6 +473,21 @@ theorem is_complemented_of_Sup_atoms_eq_top (h : Sup {a : α | is_atom a} = ⊤)
         refine hc2.directed_on.mono (λ s t, Sup_le_Sup) } },
     { rcases set.mem_sUnion.1 ha with ⟨s, sc, as⟩,
       exact (hc1 sc).2.2 a as } }
+end
+
+lemma exists_set_independent_of_Sup_atoms_eq_top (h : Sup {a : α | is_atom a} = ⊤) :
+  ∃ s : set α, complete_lattice.set_independent s ∧ Sup s = ⊤ ∧ ∀ a ∈ s, is_atom a :=
```

Comment:
Similarly here, the below more obviously says "there exists an independent set of atoms such that ..."
```suggestion
  ∃ s : set α, complete_lattice.set_independent s ∧( ∀ a ∈ s, is_atom a) ∧ Sup s = ⊤ :=
```